mlrun/model_monitoring/db/stores/sqldb/sql_store.py

```diff
@@ -256,6 +349,142 @@ def list_model_endpoints(
 
         return endpoint_list
 
+    def write_application_result(self, event: dict[str, typing.Any]):
+        """
+        Write a new application result event in the target table.
+
+        :param event: An event dictionary that represents the application result, should be corresponded to the
+                      schema defined in the :py:class:`~mlrun.common.schemas.model_monitoring.constants.WriterEvent`
+                      object.
+        """
+        self._init_application_results_table()
+
+        # First, try to update an existing result
+        application_filter_dict = self.filter_endpoint_and_application_name(
+            endpoint_id=event[
+                mlrun.common.schemas.model_monitoring.WriterEvent.ENDPOINT_ID
+            ],
+            application_name=event[
+                mlrun.common.schemas.model_monitoring.WriterEvent.APPLICATION_NAME
+            ],
+        )
+
+        application_record = self._get(
+            table=self.ApplicationResultsTable, **application_filter_dict
+        )
+        if application_record:
+            # Update an existing application result
+            self._update(
+                attributes=event,
+                table=self.ApplicationResultsTable,
+                **application_filter_dict,
+            )
+        else:
+            # Write a new application result
+            event[mlrun.common.schemas.model_monitoring.EventFieldType.UID] = (
+                uuid.uuid4().hex
+            )
```

Comment:
For enrich the new `uid` primary key column in the `app_result` table. We can't use the app name as a primary key because it's not unique. An alternative can be using app_name+endpoint_id as a composite key but I preferred to use an internal UID and the other columns (endpoint_id & app_name) as foreign keys to ensure data integrity.

Reply:
I see.
In that case, why not use the more standard auto-increment in the table:
https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.Column.params.autoincrement

Reply:
Just an update - the new `uid` column will be represented by the endpoint_id, application name and result name. With that, we can optimize the update query each time we reach the DB to update an application record.
In other words, the new `uid` is basically `<endpoint_id>_<app_name>_<result_name>`.